localstack-core/localstack/services/lambda_/event_source_mapping/pollers/sqs_poller.py

```diff
@@ -103,11 +98,11 @@ def poll_events(self) -> None:
         #  https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-pipes-sqs.html#pipes-sqs-scaling
         response = self.source_client.receive_message(
             QueueUrl=self.queue_url,
-            MaxNumberOfMessages=self.sqs_queue_parameters.get(
-                "BatchSize", DEFAULT_MAX_RECEIVE_COUNT
-            ),
+            MaxNumberOfMessages=min(self.batch_size, DEFAULT_MAX_RECEIVE_COUNT),
             MessageAttributeNames=["All"],
             MessageSystemAttributeNames=[MessageSystemAttributeName.All],
+            # Override how many messages we can receive per call
+            sqs_override_max_message_count=self.batch_size,
```

Comment:
Nice 💯 

I love that we are explicit about this override on every occasion. Please highlight that pattern at an upcoming demo 🎥  and include an example in the demo notes 🙏 . That's super helpful for the wider core engineering org 🙇